[doc][cleanup] Readme cleanup (#917)
peak3d/inputstream.adaptive: README.md

# InputStream Adaptive add-on for Kodi

> This is a [Kodi](https://kodi.tv) input stream add-on wich acts as a demuxer for segmented, multi-bitrate internet streams. The most common streaming protocols such as MPEG-DASH, HLS and Microsoft Smooth Streaming are supported.

Reviewer: Sorry last comment then I'm all good :)
`wich` typo has survived, should be `which`

Author: done

# InputStream Adaptive add-on for Kodi

This is a [Kodi](https://kodi.tv) input stream add-on wich acts as a demuxer for segmented, multi-bitrate internet streams. The most common streaming protocols such as MPEG-DASH, HLS and Microsoft Smooth Streaming are supported.

> The add-on also has support for DRM protected streams, such as Google Widevine, Microsoft PlayReady and others, for some of them are only available on specific operating systems.

Reviewer: ```suggestion
The add-on also has support for DRM protected streams, such as Google Widevine, Microsoft PlayReady and others, however some are only available on specific operating systems.
```

> The add-on can be used from all video add-ons by appropriately configuring the *ListItem* properties to instruct Kodi to play the content by using this add-on.

Reviewer 2: ```suggestion
The add-on can be used from all video add-ons by appropriately configuring the *ListItem* properties to instruct Kodi to play the content via this add-on.
```

Reviewer 3: "Via" is valid.

I think "add-on" is repeated too often, but I'm not sure how to rephrase: _The add-on can be used from all video add-ons by appropriately configuring the *ListItem* properties to instruct Kodi to play the content via this mechanism._, perhaps?

Author: i think to change with something like "To enable your video add-on to instruct Kodi to playback by using this InputStream add-on, you have to appropriately configuring the *ListItem* properties." ?

> For test purposes is possible create the STRM files, where you can add the URL for the test stream to be played and the parameters to instruct Kodi to play the content by using this add-on.

Reviewer 2: ```suggestion
For testing purposes it is possible to create the STRM files where the URL for the test stream to be played as well as the parameters to instruct Kodi to play the content using this add-on can be added.
```

Reviewer 3: _For testing purposes it is possible to create STRM files that contain the URL for the test stream as well as playback parameters to pass to Kodi._ (or similar)?

Author: changed with something similar

*Please refer to the [Wiki pages](https://github.com/xbmc/inputstream.adaptive/wiki), to read detailed instructions in how to integrate it in your add-on, how to make tests, how to build it and many other information.*

> ## Features in WIP

Reviewer 2: ```suggestion
## WIP Features
```

#### Decrypting:

> Decrypting is not implemented. But it is prepared!

Reviewer: Doesn't have to be part of this PR but thought I'd mention anyway:

`Decrypting is not implemented. But it is prepared!`

This I think was relevant back in the early days of this add-on where wvdecrypter wasn't bundled with the add-on and even at one point only existed in liberty-developer's fork - I suggest this whole paragraph is removed or rewritten, perhaps mentioning that most commonly Widevine CDM (not supplied) is required.

Reviewer 3: ... especially if the above comment about DRM is valid.

Author: now that i read again you have right i will remove all this point

This is a [Kodi](https://kodi.tv) input stream add-on wich acts as a demuxer for segmented, multi-bitrate internet streams. The most common streaming protocols such as MPEG-DASH, HLS and Microsoft Smooth Streaming are supported.

The add-on also has support for DRM protected streams, such as Google Widevine, Microsoft PlayReady and others, however some are only available on specific operating systems. To use the Google Widevine DRM on non-Android systems is required the installation of the Widevine CDM module library not included with this add-on.

> 

Reviewer 3: Use of Google Widevine DRM on non-Android systems requires the installation of the Widevine CDM module library (not included with this add-on).

> To enable your video add-on to instruct Kodi to playback by using this InputStream add-on, you have to appropriately configuring the *ListItem* properties.

Reviewer 3: To enable InputStream playback via your video add-on, you must first configure the *ListItem* properties appropriately.

> For testing purposes it is possible to create the STRM files that contains the URL for the test stream as well the playback parameters to instruct Kodi to play the content by using this InputStream add-on.

Reviewer 3: For test purposes, you can create STRM files with URLs and playback parameters to instruct Kodi to use this InputStream add-on.

> *Please refer to the [Wiki pages](https://github.com/xbmc/inputstream.adaptive/wiki), to read detailed instructions in how to integrate it in your add-on, how to make tests, how to build it and many other information.*

Reviewer 3: Please refer to the [Wiki pages](https://github.com/xbmc/inputstream.adaptive/wiki) for detailed instructions on how to integrate, build and test with your add-on.

Auto will select the best resolution matching to your videoplayer display rect without any limits.
If your display resolution is 720p, you will not be able to watch 1080p videos if there are video representations available closer to 720p.

#### TODO's:

> - Adaptive bitrate switching has been enabled on Kodi 20 but need more improvements

Reviewer: *needs

Reviewer 3: _Adaptive bitrate switching has been implemented in Kodi as of v20, but will continue to be developed over time._

- Automatic / fixed video stream selection depending on max. visible display rect (some work has to be done at the inputstream interface).
- There will be many MPEG-DASH, HSL or Smooth Stream manifest types currently not supported that must be extended.

### Notes:
- This addon uses threads to download segments. The memory consumption is the sum of single segment from each stream currently playing. Refering to known streams it is < 10MB for 720p videos.

## Acknowledgements
InputStream Adaptive exists thanks to the help of many

> * Special thanks to the author / creator [@peak3d](https://github.com/peak3d)

Reviewer 2: Missing CastagnaIT and glenn guy here :)

Author: i do not know if we should add each new developer added over time
we should therefore compile a longer list and this is not something I would like to maintain...

peak3d is mentioned because this initially this was developed by him only,
IMO i would only keep the first one and the last two points (peak3d/contributors/bento)
all others kodi devs they should be implicit, or not?
